great_ape_safe/ape_api/curve.py

```diff
@@ -87,6 +85,35 @@ def _get_n_coins(self, lp_token):
             else:
                 return registry.get_n_coins(pool)[0] # note [1] tells us if there is a wrapped coin!
 
+    def _get_coin_indices(self, lp_token, pool, asset_in, asset_out):
+        if self.is_v2:
+            coins = self._get_coins(lp_token)
+            i = None
+            j = None
+            for coin in coins:
+                if coin == asset_in:
+                    i = coins.index(coin)
+                if coin == asset_out:
+                    j = coins.index(coin)
+            assert i != None and j != None
+        else:
+            registry = self._get_registry(lp_token)
+            i, j, _ = registry.get_coin_indices(pool, asset_in, asset_out)
+        return i, j
+
+    def _get_n_coins(self, lp_token):
+        if self.is_v2:
+            # TODO
+            pass
+        else:
+            registry = self._get_registry(lp_token)
+            pool = self._get_pool_from_lp_token(lp_token)
+            if registry == self.metapool_registry:
+                return registry.get_n_coins(pool)
+            else:
+                return registry.get_n_coins(pool)[
+                    0
+                ]  # note [1] tells us if there is a wrapped coin!
```

Comment:
Not sure what other changes there are for `get_n_coins`besides the abscence of the metapool registry between v1/v2 but this worked for me
```suggestion
def _get_n_coins(self, lp_token):
        registry = self._get_registry(lp_token)
        pool = self._get_pool_from_lp_token(lp_token)
        if not self.is_v2 and registry == self.metapool_registry:
            return registry.get_n_coins(pool)
        else:
            return registry.get_n_coins(pool)[0]
```